features/cli/pod.feature

```diff
@@ -244,7 +244,7 @@ Feature: pods related scenarios
   Scenario: OCP-22283:Authentication 4.0 Oauth provider info should be consumed in a pod
     Given I have a project
     When I create a new application with:
-      | image	       | quay.io/openshifttest/hello-openshift@sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339 |
+      | image	       | quay.io/openshifttest/hello-openshift@sha256:4200f438cf2e9446f6bcff9d67ceea1f69ed07a2f83363b7fb52529f7ddd8a83 |
```

Comment:
@deepakpunia I intended to review/lgtm by comparing https://quay.io/repository/openshifttest/hello-openshift/manifest/sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339 VS https://quay.io/repository/openshifttest/hello-openshift/manifest/sha256:4200f438cf2e9446f6bcff9d67ceea1f69ed07a2f83363b7fb52529f7ddd8a83 , but both show arm64 under `ARCHITECTURE `. Then check your Runner/867542/console's env, it is not using arm64 cluster. To be safe, could you share with pass log under arm64 cluster?

Reply:
Here is the execution result from ARM env :` job/Runner/867886/console`

But i see that this test is only failing for `aws-ipi-ovn-hypershift-arm-nodepool-guest` profile but it's difficult to provision same type of cluster using same profile.

Reply:
@dpuniaredhat , hmm, indeed weird, but not sure if the new sha digest image will work in such profile?

Reply:
@deepakpunia , hmm, indeed weird, and yeah without an env we are actually not sure if it fixes. Trying to provision one: https://github.com/openshift/release/pull/44090#issuecomment-1761095379 

Also, @aleskandro , do you have any quick ideas why the 56c354e78850 fails in such profile given it passes in other arm64 profiles as RP filtered arm64 pass results https://url.corp.redhat.com/c1c73bc ?

Reply:
Hi, there's no reason it shouldn't pass... Do we have a log of the failed case?

Reply:
@aleskandro Here is the failure logs https://url.corp.redhat.com/adfafddafdaf 
Full Logs : https://url.corp.redhat.com/bafdafd
Actually it difficult to find out the root cause because test failing intermittently

Reply:
@dpuniaredhat the case you are running is on an x86_64 hypershift management cluster with an aarch64 node pool (guest cluster).

The fact it is failing intermittently seems like there is some other non-aarch64 node where the pod can run.

In those cases, what can happen is that the ImageStreamTag being imported due to the `new-app` command will unpack the manifest list and import the manifest according to the architecture of the node hosting the internal registry (e.g., amd64).

Then, when we create a deployment running that image, that deployment could be scheduled on a node of a different arch (e.g., arm64), hence the failure.

@LiangquanLi930 can you confirm this profile is x86_64 mgmt + aarch64 hosted or x86_64 mgmt + (aarch64 + x86_64) hosted?

Now, for OCP >=4.14, you can add `--import-mode=PreserveOriginal` to fix this case in the `I create a new application with` action. This way, the image will be imported as ManifestList, and whatever the architecture of the node where the pod is scheduled, it will run fine.

Reply:
@aleskandro @deepakpunia https://github.com/openshift/release/pull/44090#issuecomment-1761095379 env is ready, guest and management kubeconfigs are https://url.corp.redhat.com/xxia-shared-google-doc , if you want to have a live env to check, could you check?

Reply:
Apologize, new-app bases its work on the architecture where `oc` runs. Therefore, although the image registry is running on arm64, we import the amd64 image.

The following, instead, works for oc>=1.14
```
oc new-app --image=quay.io/openshifttest/hello-openshift@sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339 --labels=name\=ruby-ex  --import-mode=PreserveOriginal
```

Reply:
```
└ $ oc new-app --image=quay.io/openshifttest/hello-openshift@sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339 --labels=name\=ruby-ex 
--> Found container image e4f3998 (16 months old) from quay.io for "quay.io/openshifttest/hello-openshift@sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339"

    * An image stream tag will be created as "hello-openshift:latest" that will track this image

--> Creating resources with label name=ruby-ex ...
    imagestream.image.openshift.io "hello-openshift" created
    deployment.apps "hello-openshift" created
    service "hello-openshift" created
--> Success
    Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
     'oc expose service/hello-openshift' 
    Run 'oc status' to view your app.
┌ [Fri Oct 13 12:27:48 CEST 2023] 12:27:48 up 1 day, 23:55,  6 users,  load average: 1.51, 1.24, 1.40 mem usage: 0.70 root disk: 0.11
├ aleskandro@mifune ~/__hot_repos/oc (⎈|my-test/ab89017db966e48e9979a6e67729efbd-e99c6ef098b9fdc3-elb-us-east-1-amazonaws-com:6443/system:admin:my-test) (master*?)  
└ $ oc get pods
NAME                               READY   STATUS   RESTARTS     AGE
hello-openshift-66bdb89f6c-6sjfg   0/1     Error    1 (2s ago)   2s
┌ [Fri Oct 13 12:27:50 CEST 2023] 12:27:50 up 1 day, 23:55,  6 users,  load average: 1.51, 1.24, 1.40 mem usage: 0.70 root disk: 0.11
├ aleskandro@mifune ~/__hot_repos/oc (⎈|my-test/ab89017db966e48e9979a6e67729efbd-e99c6ef098b9fdc3-elb-us-east-1-amazonaws-com:6443/system:admin:my-test) (master*?)  
└ $ oc new-project test-2                                     
Now using project "test-2" on server "https://ab89017db966e48e9979a6e67729efbd-e99c6ef098b9fdc3.elb.us-east-1.amazonaws.com:6443".

You can add applications to this project with the 'new-app' command. For example, try:

    oc new-app rails-postgresql-example

to build a new example application in Ruby. Or use kubectl to deploy a simple Kubernetes application:

    kubectl create deployment hello-node --image=registry.k8s.io/e2e-test-images/agnhost:2.43 -- /agnhost serve-hostname

┌ [Fri Oct 13 12:28:08 CEST 2023] 12:28:08 up 1 day, 23:56,  6 users,  load average: 1.16, 1.17, 1.37 mem usage: 0.70 root disk: 0.11
├ aleskandro@mifune ~/__hot_repos/oc (⎈|test-2/ab89017db966e48e9979a6e67729efbd-e99c6ef098b9fdc3-elb-us-east-1-amazonaws-com:6443/system:admin:test-2) (master*?)  
└ $ /tmp/bisect/oc new-app --image=quay.io/openshifttest/hello-openshift@sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339 --labels=name\=ruby-ex  --import-mode=PreserveOriginal
--> Found container image 56c354e (1 second old) from quay.io for "quay.io/openshifttest/hello-openshift@sha256:56c354e7885051b6bb4263f9faa58b2c292d44790599b7dde0e49e7c466cf339"

    * An image stream tag will be created as "hello-openshift:latest" that will track this image

--> Creating resources with label name=ruby-ex ...
    imagestream.image.openshift.io "hello-openshift" created
    deployment.apps "hello-openshift" created
    service "hello-openshift" created
--> Success
    Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
     'oc expose service/hello-openshift' 
    Run 'oc status' to view your app.
┌ [Fri Oct 13 12:28:25 CEST 2023] 12:28:25 up 1 day, 23:56,  6 users,  load average: 1.12, 1.16, 1.37 mem usage: 0.70 root disk: 0.11
├ aleskandro@mifune ~/__hot_repos/oc (⎈|test-2/ab89017db966e48e9979a6e67729efbd-e99c6ef098b9fdc3-elb-us-east-1-amazonaws-com:6443/system:admin:test-2) (master*?)  
└ $ oc get pods
NAME                               READY   STATUS    RESTARTS   AGE
hello-openshift-578fdd579f-9l9dm   1/1     Running   0          2s
```

Reply:
@aleskandro Is this tag  `--import-mode=PreserveOriginal` will work for all release or only `for >4.14`?

Reply:
only `>=4.14`.

Reply:
@aleskandro , it is propagated to `@openshift-qe` in Slack? If not, seems it is worthy to, many subteams use new-app in auto but maybe not know this tricky profile, could you spread it?

Reply:
Hi @xingxingxia i actually prefer to resolve this problem at the base by changing the `I create a new application with` action.

I've seen some PR recently that could have done that. @liangxia you were assigned to it iirc.

Reply:
We have specific steps for this issue already, refer to https://github.com/openshift/verification-tests/pull/3466 for examples.

Reply:
I wonder if we could remove the ` importmode if supported with` and fix this problem once and for all.

Reply:
@liangxia I've commented on that PR which is referencing a step that doesn't exist...so https://github.com/openshift/cucushift/pull/9650 is invalid.

Reply:
Let me know if this works for you or not, https://github.com/openshift/verification-tests/pull/3468

Reply:
@liangxia yes, it does.

Reply:
@xingxingxia As it's fixed at framework level  https://github.com/openshift/verification-tests/pull/3468, Can we close this PR?